### DISCUSSION TITLE
Pbend (Pitch Bend) Range

### DISCUSSION THREAD

**Discussion by garamoniker** [GeekFunkLabs/fluidpatcher#48] [Q&A] [answered]
Hi guys,

Apologies, I promise this will be my last rookie question...for a while. 

I want to increase the pitch bend range to a full octave. I've blindly tried:

- {type: pbend, chan: 1=2-6, par: 12} 

But, of course, to no avail. Any help would be greatly appreciated. Thanks again!

**Comment by albedozero** [comment, marked as answer]
All levels of question are welcome! That's totally what this forum is for.

You can change the pitch bend range using a registered parameter number (RPN). These are special control change messages you can send to modify synthesizer parameters, and they're part of the MIDI specification. Put the following in a `messages` block in your bank file:
```yaml
messages:
- cc:1:101:0
- cc:1:100:0
- cc:1:6:12
```
The first two messages select the pitch bend range RPN on MIDI channel 1, then the third sets it to 12 semitones (a full octave). I'm working on a lesson video about all the things you can do with RPNs and NRPNs, but this [blog post](https://www.recordingblogs.com/wiki/midi-registered-parameter-number-rpn) has a good explanation.

**Reply by garamoniker** [reply]
Thanks again for your sagacity, Bill. I presume this can nestle comfortably beneath router rules, above the patches?

**Reply by albedozero** [reply]
Yes - in fact, the ordering of `router_rules`, `messages`, `patches`, `fluidsettings` etc. doesn't matter, except that if you have more than one (e.g. two `messages` blocks) only the last one will be used.

**Reply by garamoniker** [reply]
Ah! Okay. So, for instance, with the demonstration file you've provided with the Squishbox setup, you have: 

init:
  messages: [cc:4:37:60, cc:4:65:127, cc:4:68:127,
             cc:5:71:0, cc:5:73:0, cc:5:74:0, cc:5:78:40, cc:5:79:40, cc:5:80:80, cc:5:81:127, cc:5:82:80, 
             cc:5:71:0, cc:5:73:0, cc:5:74:0, cc:5:78:40, cc:5:79:40, cc:5:80:80, cc:5:81:127, cc:5:82:80, 
             cc:6:100:0, cc:6:101:0, cc:6:6:12, cc:6:38:0]

At the bottom of the script. So, if I just slap it in there, like this (please see last three messages added):

init:
  messages: [cc:4:37:60, cc:4:65:127, cc:4:68:127,
             cc:5:71:0, cc:5:73:0, cc:5:74:0, cc:5:78:40, cc:5:79:40, cc:5:80:80, cc:5:81:127, cc:5:82:80, 
             cc:5:71:0, cc:5:73:0, cc:5:74:0, cc:5:78:40, cc:5:79:40, cc:5:80:80, cc:5:81:127, cc:5:82:80, 
             cc:6:100:0, cc:6:101:0, cc:6:6:12, cc:6:38:0, cc:1:101:0, cc:1:100:0, cc:1:6:12]

**Reply by albedozero** [reply]
Exactly. Also, by putting it in the`init` block it just gets set once, when the bank is loaded.

**Reply by garamoniker** [reply]
Awesome. Thanks so much, Bill.
